### DISCUSSION TITLE
Are actions protected if their loaders are protected?

### DISCUSSION THREAD

**Discussion by vedantroy** [remix-run/remix#2846] [Q&A] [answered]
Let's say I have a loader that checks whether a user is authenticated.
If the loader denies access b/c the user is not authenticated, would someone theoretically be able to fire-up Postman & still make a POST Request to the action in the route file?

**Comment by sergiodxa** [comment, marked as answer]
When Remix receives a GET /some/url request it calls the loader, and when it receives a POST /some/url request it calles the action (also for other non-GET methods).

So if you want to protect a route that has both loader and actions, you must add your route protection on both the loader and action.
